From 4f735f5e204e19fa6175c15482db46b2c92ea3c0 Mon Sep 17 00:00:00 2001
From: goetz <goetz@web-dev.de>
Date: Wed, 8 Sep 2021 18:51:33 +0200
Subject: [PATCH] Invalid pointer access

If a vector is empty, accessing a vector first element within
dereferencing the first element
is access to invalid memory.
---
 lib-src/zipios++/zipios++/zipheadio.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib-src/zipios++/zipios++/zipheadio.h b/lib-src/zipios++/zipios++/zipheadio.h
index 101443a1..25be129b 100644
--- a/lib-src/zipios++/zipios++/zipheadio.h
+++ b/lib-src/zipios++/zipios++/zipheadio.h
@@ -150,7 +150,7 @@ inline void readByteSeq ( istream &is, vector < unsigned char > &vec, int count
 }
 
 inline void writeByteSeq ( ostream &os, const vector < unsigned char > &vec ) {
-  os.rdbuf()->sputn( reinterpret_cast< const char * >( &( vec[ 0 ] ) ), vec.size() ) ;
+  os.rdbuf()->sputn( reinterpret_cast< const char * >( vec.data() ), vec.size() ) ;
 }
 
 istream& operator>> ( istream &is, ZipLocalEntry &zlh         ) ;
-- 
2.34.1

